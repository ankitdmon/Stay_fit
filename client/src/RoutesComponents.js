import { Navigate, Route, Routes } from "react-router-dom";
import Meals from './MealsComponents/Meals';
import BMICalculator from './BMRCalculator';
import Login from './Login'
import Signup from './Signup';
import Home from './Home';
import FoodJournal from './MealsComponents/FoodJournal';
import PrivateRoutes from "./PrivateRoutes";

function RoutesComponents({ login, signUp }) {
    return (
        <Routes>
            <Route exact path="/" element={<Home />}></Route>
            <Route exact path="/login" element={<Login login={login} />}></Route>
            <Route exact path="/signup" element={<Signup signUp={signUp} />}></Route>
            <Route path="/bmr"
                element={
                    <BMICalculator />
                }
            />
            <Route element={<PrivateRoutes />}>
                <Route path="/meals"
                    element={
                        <Meals />
                    }
                />
                <Route path="/foodJournal"
                    element={
                        <FoodJournal />
                    }
                />
            </Route >
            <Route path="*" element={<Navigate to="/" />} />
        </Routes>

    )
}

export default RoutesComponents;