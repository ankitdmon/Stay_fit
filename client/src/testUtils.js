import React from "react";
import UserContext from "./UserContext";

const demoUser = {
    username: "testuser",
    first_name: "testfirst",
    last_name: "testlast",
    email: "test@test.net",
};

const UserProvider =
    ({ children, currentUser = demoUser }) => (
        <UserContext.Provider value={{ currentUser }}>
            {children}
        </UserContext.Provider>
    );

export { UserProvider };
