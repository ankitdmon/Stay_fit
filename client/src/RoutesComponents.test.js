
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { UserProvider } from "./testUtils";
import RoutesComponents from "./RoutesComponents";

it("renders without crashing", function () {
    render(
        <MemoryRouter>
            <UserProvider>
                <RoutesComponents />
            </UserProvider>
        </MemoryRouter>,
    );
});

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <RoutesComponents />
            </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});
