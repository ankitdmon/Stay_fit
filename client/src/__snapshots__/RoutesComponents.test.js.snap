// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h2
    class="mb-4 font-weight-bold homeTitle"
  >
    Welcome Back, testuser !
    <p
      class="quote"
    >
      To ensure good health: eat lightly, breathe deeply, live moderately, cultivate cheerfulness, and maintain an interest in life
    </p>
  </h2>
</DocumentFragment>
`;
